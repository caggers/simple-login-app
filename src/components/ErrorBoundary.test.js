import React from 'react';
import { shallow, mount } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';

function ProblemChild() {
  throw new Error('Error thrown from problem child');
  return <h1>Something went wrong</h1>; //eslint-disable-line
}
describe('<ErrorBoundary />', () => {
  const wrapper = shallow(<ErrorBoundary />);
  it('displays an error message on error generated by child', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('initialises a `state` with hasError which is false', () => {
    expect(wrapper.state()).toEqual({
      hasError: false,
      error: null,
      errorInfo: null,
    });
  });

  describe('#componentDidCatch', () => {
    // pare down the crazy console output which is spat out as a result of causing an error on purpose
    jest.spyOn(window._virtualConsole, 'emit').mockImplementation(() => false);

    const spy = jest.spyOn(ErrorBoundary.prototype, 'componentDidCatch');
    const mounted = mount(
      <ErrorBoundary>
        <ProblemChild />
      </ErrorBoundary>
    );

    it('should catch errors with `componentDidCatch`', () => {
      expect(spy).toHaveBeenCalledTimes(1);
    });

    it('should set the `state.hasError` to true in componentDidCatch', () => {
      expect(mounted.state().hasError).toEqual(true);
    });
  });
});
