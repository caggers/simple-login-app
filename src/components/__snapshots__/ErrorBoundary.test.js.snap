// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> displays an error message on error generated by child 1`] = `""`;
